Resources/public/js/views/fields/ez-relationlist-editview.js

```diff
@@ -129,20 +129,37 @@ YUI.add('ez-relationlist-editview', function (Y) {
          */
         _variables: function () {
             var relatedContents = this.get('relatedContents'),
-                relatedContentsJSON = [];
+                relatedContentsJSON = [],
+                isNewRelation = Y.bind(this._isNewRelation, this);
 
             Y.Array.each(relatedContents, function (value) {
-                relatedContentsJSON.push(value.toJSON());
+                var relatedContentJSON = value.toJSON();
+
+                if (isNewRelation(value) || relatedContentJSON.resources.MainLocation) {
```

Comment:
should this be `this._ isNewRelation(value)` ?

Reply:
True, it could be simplified by passing `this` as the third argument of `Y.Array.each` 😉